src/battle_ai_main.c

```diff
@@ -819,7 +819,22 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
         break;
     case AI_EFFECTIVENESS_x0_125:
     case AI_EFFECTIVENESS_x0_25:
```

Comment:
I would just remove those cases since their are already properly handled during ai damage calcs.

Reply:
Low damage does not reduce score and will not cause the AI to switch pokemon, just to select the most damaging move.

Reply:
We don't need to reduce score on low damage. The best damaging move will always get a +1

Reply:
If all moves have low damage the AI won't switch out because the scores are still at 100.

Reply:
oh I see the problem. I don't think this is well handled (switching should be based of damage not by score) but it's fine.